test/integration/component/test_router_resources.py

```diff
@@ -0,0 +1,824 @@
+
+        # Create new domain1
+        cls.domain1 = Domain.create(
+            cls.apiclient,
+            services=cls.services["acl"]["domain1"],
+            parentdomainid=cls.domain.id)
+
+        # Create account1
+        cls.account1 = Account.create(
+            cls.apiclient,
+            cls.services["acl"]["accountD1"],
+            domainid=cls.domain1.id
+        )
+
+        # Create Network Offering with all the services
+        cls.network_offering = NetworkOffering.create(
+            cls.apiclient,
+            cls.services["isolated_network_offering"]
+        )
+        # Enable Network offering
+        cls.network_offering.update(cls.apiclient, state='Enabled')
+
+        cls.network = Network.create(
+            cls.apiclient,
+            cls.services["isolated_network"],
+            accountid=cls.account1.name,
+            domainid=cls.account1.domainid,
+            networkofferingid=cls.network_offering.id,
+            zoneid=cls.zone.id
+        )
+
+        virtualmachine = VirtualMachine.create(
+            cls.apiclient,
+            services=cls.services["virtual_machine_userdata"],
+            accountid=cls.account1.name,
+            domainid=cls.account1.domainid,
+            serviceofferingid=cls.service_offering.id,
+            networkids=cls.network.id,
+            templateid=cls.template.id,
+            zoneid=cls.zone.id
+        )
+
+        vms = list_virtual_machines(
+            cls.apiclient,
+            account=cls.account1.name,
+            domainid=cls.account1.domainid,
+            id=virtualmachine.id
+        )
+        vm = vms[0]
+
+        # get vm cpu and memory values
+        cls.vm_cpu_count = vm.cpunumber
+        cls.vm_mem_count = vm.memory
+
+        routers = list_routers(
+            cls.apiclient,
+            account=cls.account1.name,
+            domainid=cls.account1.domainid
+        )
+
+        router = routers[0]
+
+        router_so_id = router.serviceofferingid
+
+        list_service_response = list_service_offering(
+            cls.apiclient,
+            id=router_so_id,
+            issystem="true",
+            systemvmtype="domainrouter",
+            listall="true"
+        )
+
+        # Get default router service offering cpu and memory values
+        cls.default_vr_cpu = list_service_response[0].cpunumber
+        cls.default_vr_ram = list_service_response[0].memory
+
+        cls._cleanup.append(virtualmachine)
+        cls._cleanup.append(cls.network)
+
+        # Disable Network offering
+        cls.network_offering.update(cls.apiclient, state='Disabled')
+        cls._cleanup.append(cls.network_offering)
+        cls._cleanup.append(cls.account1)
+        cls._cleanup.append(cls.domain1)
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
```

Comment:
```suggestion
        super(TestRouterResources, cls).tearDownClass()
```

Reply:
@CharlesQueiroz the parent class implements a cleanup of a reverse list of added objects and does some extra for some types. Please use it.